isobus/include/isobus/isobus/can_parameter_group_builder.hpp

```diff
@@ -0,0 +1,503 @@
+#pragma once
+
+#include <vector>
+
+namespace isobus
+{
+	// Note that this class currently only works for packets eight bytes or fewer, because that is
+	// all the use-cases I needed.  The spec has a whole section on how data can span byte
+	// boundaries, and how to align the bits when a data-type that is not a multiple of eight bits
+	// crosses said boundaries.  Then every single parameter group uses padding to avoid those
+	// cases!  The five-bit tractor command types have three bits of padding to make them exactly
+	// one byte.  Why even specify the scheme if it isn't used?
```

Comment:
They do a good job of padding, that is true, but I have often seen SPNs span bytes in an unaligned fashion in proprietary messaging, so it is more common than you might think

Reply:
This is more a rant than anything.  the code does handle byte spans correctly (at least I believe it does).